Reentry-To-Recovery/r2r-frontend
# Feature/RTR-40: Admin Lessons

## Item Tracking

Jira link: https://reentrytorecovery.atlassian.net/browse/RTR-40

## Description

This PR implements admin add lesson, edit lesson, delete lesson, and order course section content.

## Motivation

We're doing this because admin users will need to manage lessons.

## Test Plan

- Log in as an admin user
- Navigate to the dashboard and go to edit a course
- Expand the course builder
- Expand a topic
- Under the topic, you should see the lessons and quizzes and add lesson and add quiz buttons

### Add Lesson
- Click on the add lesson button
- An add lesson modal should open
- The button should initially be disabled, and does not become enabled until all fields are valid
- Try to add a new lesson
- Upon successfully submitting, the modal should close and you should see a success toast
- You should see the new course

### Edit Lesson
- To edit a lesson, click on the pen icon
- An edit lesson modal should open pre-filled
- The button should be initially be disabled, and doesn't become enabled until changes have made and those changes are valid
- Upon successfully editing, the modal should close and you should see a success toast
- You should see the updated title if the title is updated

### Delete Lesson
- To delete a lesson, click on the delete icon
- You should see a confirmation
- Upon successful deletion, you should see a success toast
- You should no longer see the deleted lesson

### Order Course Section Content
- Drag the course section content to reorder
- Refresh the page
- The content should be in the new order

## Further Comments

The edit button will need to be updated to account for edit quiz vs lesson
